📝 Add Documentation for Call Activity (#7)
atlas-engine/ContributionGuidelines: docs/getting_started/bpmn_elements/call_activity.md

# Call Activity

> ![Call Activity](../images/call_activity.svg)

Reviewer: Die Einstellungen im Property Panel sollten hier ebenfalls gezeigt werden. 

Am besten vollständig mit Payload und StartEvent ID (auch wenn diese optional sind).

Auch würde ich hier vllt einen Beispielnamen einsetzen; "Sample Call Activity" oder sowas.

<img width="350px" src="../images/call_activity_property-panel.png">

> The Call Activity is used to execute other processes within a process.

Reviewer: ```suggestion
The Call Activity is used to execute other processes as a subprocess.
```

> When the Call Activity is reached, the Call Activity is in a suspended state until the referenced process is terminated or runs into an error.

Reviewer: ```suggestion
When the Call Activity is reached, the Call Activity is in a suspended state until the referenced process has finished.
```

Wie der Subprozess hier beendet wird ist egal, die CallActivity geht so oder so erst weiter, wenn der Subprozess fertig ist.

> When the call activity is reached, the call activity is in a suspended state until the referenced process is terminated or runs into an error.

Reviewer 2: ```suggestion
When the call activity is reached, the call activity is in a suspended state until the referenced process has been finished or runs into an error.
```

Author: Ich finde `is terminated` passender, da es weniger spezifisch ist. Wenn z.B. der Referenzierte Prozess gestoppt wird bleibt die CallActivity ja auch nicht suspended sondern wird beendet.

Reviewer: `...referenced process is terminated or runs into an error.` impliziert, dass ein Subprozess immer nur terminiert wird oder in einen Error läuft. Nicht aber, dass der auch einfach ganz normal fertig wird. 

Insofern ist "finished" hier die bessere Wahl. :+1:

> When the Call Activity is reached, the Call Activity is in a suspended state until the referenced process has finished.

Reviewer: ```suggestion
When the Call Activity is reached, the Call Activity is suspended, until the referenced process has finished.
```

> If the referenced process runs into an error the Call Activity will also fail with the same error.

Reviewer: ```suggestion
If the referenced process runs into an error, the Call Activity will also fail with the same error.
```

> If the referenced process runs into an error, the Call Activity will also fail with the same error.

Reviewer: ```suggestion
If the referenced process runs into an error, the Call Activity also fails with the same error.
```

> There are two things that need to be configured:

Reviewer: Ich würde diese Sektion in Pflichtfelder und optionale Felder aufteilen. 

Ungefähr so:

```md
## Configuration

Required parameters
- `calledReference`: The ID of the Process Model to call

Optional Parameters:
- `startEventId`: The ID of the Start Event at which to start the subprocess. If not provided, the first available Start Event is used
- `payload`: The payload with which to start the subprocess. If not provided, the current process token is used
```

> 1. ProcessDefinitionId

Reviewer: Nope.

```suggestion
1. ProcessModelId
```

2. StartEventId

The ProecssDefinitionId is used to define the diagram that should get executed.

> The StartEventId is used to define which start event of the given diagram should be used when executing the diagram.

Reviewer: Hierbei sollte beachtet werden, dass die StartEvent ID optional ist. Wenn keine angegeben ist, wird einfach die erste genommen, die an dem Diagramm modelliert ist.

Ebenfalls sollte hier der Payload noch dokumentiert werden.
